packages/web3-utils/src/converters.ts

```diff
@@ -0,0 +1,340 @@
+		throw new Error(errorMessage(str, 'not a valid string'));
+	}
+
+	// To be compatible with 1.x trim null character
+	// eslint-disable-next-line no-control-regex
+	let strWithoutNullCharacter = str.replace(/^(?:\u0000)/, '');
+	// eslint-disable-next-line no-control-regex
+	strWithoutNullCharacter = strWithoutNullCharacter.replace(/(?:\u0000)$/, '');
+
+	return `0x${Buffer.from(strWithoutNullCharacter, 'utf8').toString('hex')}`;
+};
+
+/**
+ * @alias `utf8ToHex`
+ */
+
+export const fromUtf8 = utf8ToHex;
+/**
+ * @alias `utf8ToHex`
+ */
+export const stringToHex = utf8ToHex;
+
+/**
+ * Should be called to get utf8 from it's hex representation
+ */
+export const hexToUtf8 = (str: HexString): string => bytesToBuffer(str).toString('utf8');
+
+/**
+ * @alias `hexToUtf8`
+ */
+export const toUtf8 = hexToUtf8;
+
+/**
+ * @alias `hexToUtf8`
+ */
+export const hexToString = hexToUtf8;
+
+/**
+ * Should be called to get hex representation (prefixed by 0x) of ascii string
+ */
+export const asciiToHex = (str: string): HexString => {
+	if (typeof str !== 'string') {
+		throw new Error(errorMessage(str, 'not a valid string'));
+	}
+
+	return `0x${Buffer.from(str, 'ascii').toString('hex')}`;
+};
+
+/**
+ * @alias `asciiToHex`
+ */
+export const fromAscii = utf8ToHex;
+
+/**
+ * Should be called to get ascii from it's hex representation
+ */
+export const hexToAscii = (str: HexString): string => bytesToBuffer(str).toString('ascii');
+
+/**
+ * @alias `hexToAscii`
+ */
+export const toAscii = hexToUtf8;
+
+/**
+ * Auto converts any given value into it's hex representation.
+ */
+export const toHex = (
+	value: Numbers | Bytes | Address | boolean,
+	returnType?: boolean,
+): HexString | ValueTypes => {
+	if (typeof value === 'string' && isAddress(value)) {
+		return returnType ? 'address' : `0x${value.toLowerCase().replace(/^0x/i, '')}`;
+	}
+
+	if (typeof value === 'boolean') {
+		// eslint-disable-next-line no-nested-ternary
+		return returnType ? 'bool' : value ? '0x01' : '0x00';
+	}
+
+	if (typeof value === 'number') {
+		// eslint-disable-next-line no-nested-ternary
+		return returnType ? (value < 0 ? 'int256' : 'uint256') : numberToHex(value);
+	}
+
+	if (typeof value === 'bigint') {
+		return returnType ? 'bigint' : numberToHex(value);
+	}
+
+	if (typeof value === 'string') {
+		if (value.startsWith('-0x') || value.startsWith('-0X')) {
+			return returnType ? 'int256' : numberToHex(value);
+		}
+
+		if (value.startsWith('0x') || value.startsWith('0X')) {
+			return returnType ? 'bytes' : value;
+		}
+
+		if (!Number.isFinite(value)) {
+			return returnType ? 'string' : utf8ToHex(value);
+		}
+	}
+
+	throw new Error(errorMessage(value, 'can not be converted to hex'));
+};
+
+/**
+ * Auto converts any given value into it's hex representation,
+ * then converts hex to number.
+ */
+export const toNumber = (value: Numbers): number | bigint => {
+	if (typeof value === 'number' || typeof value === 'bigint') {
+		return value;
+	}
```

Comment:
What do you think about this?

```suggestion
	if (typeof value === 'number') return value;
	
        if (typeof value === 'bigint') {
                return value >= Number.MIN_SAFE_INTEGER || value <= Number.MAX_SAFE_INTEGER ?
                        Number(value) : value;
	}
```